operator_ui/src/Private.tsx

```diff
@@ -0,0 +1,241 @@
+import Header from 'containers/Header'
+import Loading from 'components/Loading'
+import Notifications from 'containers/Notifications'
+import { ShowComponentProps } from 'containers/Jobs/Show'
+import PrivateRoute from './PrivateRoute'
+
+// Asynchronously load routes that are chunked via code-splitting
+// 'import' as a function must take a string. It can't take a variable.
+
+const uniOpts = { loading: Loading }
+const DashboardsIndex: UniversalComponent<Pick<
+  {
+    classes: any
+  },
+  never
+> &
+  StyledComponentProps<'wrapper' | 'text'> & {
+    recentJobRunsCount: number
+    recentlyCreatedPageSize: number
+  }> = universal(import('./containers/Dashboards/Index'), uniOpts)
+const JobsIndex = universal(import('./containers/Jobs/Index'), uniOpts)
+const JobsShow: UniversalComponent<ShowComponentProps> &
+  StyledComponentProps<'wrapper' | 'text'> = universal(
+  import('./containers/Jobs/Show'),
+  uniOpts,
+)
+const JobsDefinition = universal(
+  import('./containers/Jobs/Definition'),
+  uniOpts,
+)
+const JobsErrors = universal(import('./containers/Jobs/Errors'), uniOpts)
+const JobsNew = universal(import('./containers/Jobs/New'), uniOpts)
+const BridgesIndex = universal(import('./containers/Bridges/Index'), uniOpts)
+const BridgesNew = universal(import('./containers/Bridges/New'), uniOpts)
+const BridgesShow = universal(import('./containers/Bridges/Show'), uniOpts)
+const BridgesEdit = universal(import('./containers/Bridges/Edit'), uniOpts)
+const JobRunsIndex: UniversalComponent<Pick<
+  {
+    classes: any
+  },
+  never
+> &
+  StyledComponentProps<'wrapper' | 'text'> & {
+    pagePath: string
+  }> = universal(import('./containers/JobRuns/Index'), uniOpts)
+const JobRunsShowOverview = universal(
+  import('./containers/JobRuns/Show/Overview'),
+  uniOpts,
+)
+const JobRunsShowJson = universal(
+  import('./containers/JobRuns/Show/Json'),
+  uniOpts,
+)
+const JobRunsShowErrorLog = universal(
+  import('./containers/JobRuns/Show/ErrorLog'),
+  uniOpts,
+)
+const TransactionsIndex = universal(
+  import('./containers/Transactions/Index'),
+  uniOpts,
+)
+const TransactionsShow = universal(
+  import('./containers/Transactions/Show'),
+  uniOpts,
+)
+const Configuration = universal(
+  import('./containers/Configuration/Index'),
+  uniOpts,
+)
+const NotFound = universal(import('./containers/NotFound'), uniOpts)
+
+const styles = (theme: Theme) => {
+  return {
+    content: {
+      marginTop: 0,
+      marginBottom: theme.spacing.unit * 5,
+    },
+  }
+}
+
+const Private = ({ classes }: { classes: { content: string } }) => {
+  const [headerHeight, setHeaderHeight] = React.useState(0)
+
+  let drawerContainer: HTMLElement | null = null
+
+  return (
+    <Grid container spacing={0}>
+      <Grid item xs={12}>
+        <Header
+          onResize={(_width, height) => setHeaderHeight(height)}
+          drawerContainer={drawerContainer}
+        />
+        <main
+          ref={ref => {
+            drawerContainer = ref
```

Comment:
Im not sure about the context of the functionality but shouldn't we use createRef/useRef here?

Reply:
Good shout, we should

Reply:
Actually, had to revert back, because `@material-ui` lib did not expect a ref object type.